app/src/server.js

```diff
@@ -78,11 +78,22 @@ app.use(passport.initialize());
 
 app.get("/login", spidAuth);
 
-app.post("/assertionConsumerService", spidAuth, function(
+app.post("/assertionConsumerService", function(
   req: express$Request,
-  res: express$Response
+  res: express$Response,
+  next: express$NextFunction
 ) {
-  acsController.acs(req, res);
+  passport.authenticate("spid", function(err, user) {
```

Comment:
this function is constant and can be declared outside of the scope of this callback:

``` js 
const myFunction = passport.authenticate(....);
app.post("...", myFunction);
``` 

also this is hard to read because `acsController` is hardcoded inside the authentication code when in fact it should be composed with, you can do something like this, that is more readable:

``` js
// decompose the spid authorization from what happens with the user
function withSpidAuth(controller: (user, req, res, next) => any): (req, res, next) => any {
  return passport.authenticate("spid", function(err, user) {
  // ... failure checks ...
  controller(user, req, res, next);
  });
}

// then...

app.post("...", withSpidAuth(acsController.acs));

```